### DISCUSSION TITLE
want to know about upgrade

### DISCUSSION THREAD

**Discussion by Srashti-jain** [patternfly/patternfly#5812] [General]
Hi team, 
I want to know how to upgrade Patternfly 4 to Patternfly 5 in my existing project.
Please help me step by step.

**Comment by nicolethoen** [comment]
Hello @Srashti-jain! we are excited to see people migrating!

We do have a guide for upgrading on our website: https://www.patternfly.org/get-started/upgrade
As you work through this guide, please feel free to reach out with any questiosn!

**Comment by Srashti-jain** [comment]
I read the upgrade documentation but I am confused.
Can you explain to me step by step?

**Reply by nicolethoen** [reply]
Do you already have an app running with PatternFly 4 dependencies?

**Comment by Srashti-jain** [comment]
Yes, I have! I am working at an organization and we are using patternfly in
our running project.

**Comment by nicolethoen** [comment]
So i'd recommend updating the versions of the PatternFly dependencies in your project's package.json file to the promoted versions listed at the top of this page: https://www.patternfly.org/get-started/release-highlights
after that's done, from your project's root directory run this command `npx @patternfly/pf-codemods@latest <path to your source code>`

This will output in your console all the changes you'll need to make to consume PF5. Browse the list and if you're comfortable, you can run it again but add a `--fix` flag so the code mods will make a suite of changes for you. Then you can read the remaining errors flagged in the console and address them one at a time manually. That's all explained in more detail [here](https://www.patternfly.org/get-started/upgrade#run-our-codemods)

Hopefully that will get your code base building/compiling again. Then you'll want to run the app and see how things are appearing visually. If there are many CSS style overrides written on top of PF, likely they will need to be updated according to [these instructions](https://www.patternfly.org/get-started/upgrade#review-and-update-variable-and-class-names).

Hopefully that helps!
